### ISSUE TITLE
Segmentation fault (core dumped) when performing sampling many times

### ISSUE THREAD

**Issue by jordai** [cnr-isti-vclab/PyMeshLab#216]
First of all, thanks for the great library, it is of great help in my Master's thesis on Single-View 3D Reconstruction.

For my project, I am training a Neural Network to perform 3D Reconstruction given a single image.
For one of the loss functions, I want to sample vertices on the surface of a predicted mesh.

First, I tried to to Poisson Disk Sampling, since I used that for generating the dataset as well.
This led to an error that said: ```segmentation fault (core dumped)```.
Besides this, it also prints a ```UserWarning: semaphore_tracker: There appear to be 6 leaked semaphores to clean up at shutdown```.

After tracking the problem down, I noticed that applying the Poisson Disk Sampling filter results in a monotonically increasing RAM usage, until it reaches a maximum capacity of memory available.
Therefore, I tried a different sampling method (Monte Carlo), which seemed to perform better.
After a longer time of training, I get the same error as with the Poisson Disk Sampling.

Do you know how or why this happens? 
It seems to me that sampling should not monotonically increase memory usage.
I included a piece of code that results in this error, including the vertices and faces files.

I am using PyMeshLab 2022.2.post2, and Python 3.7.
I run my code on Ubuntu 20.04.

```
import pymeshlab as ml
import numpy as np

vertices = np.loadtxt('pred_verts.txt')
faces = np.loadtxt('faces.txt', dtype='int')
ms = ml.MeshSet()

for i in range(0, 119900):

    mesh = ml.Mesh(vertex_matrix=vertices, face_matrix=faces)
    ms.add_mesh(mesh)
    ms.apply_filter('poisson_disk_sampling', samplenum=3000,
                    radius=ml.Percentage(0),
                    montecarlorate=20, approximategeodesicdistance=False,
                    subsample=False,
                    refineflag=False)
    ms.clear()
```
[faces.txt](https://github.com/cnr-isti-vclab/PyMeshLab/files/8659428/faces.txt)
[pred_verts.txt](https://github.com/cnr-isti-vclab/PyMeshLab/files/8659429/pred_verts.txt)

**Comment by matfcp** [comment]
I'm facing the same issue with Poisson Disk Sampling as well. I'm generating point clouds for training neural networks. After repeatedly calling this filter in a script, it shows the message "segmentation fault (core dumped)".
